net-irc/znc/files/znc.initd-r1

```diff
@@ -0,0 +1,40 @@
+#!/sbin/runscript
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+extra_commands="config"
+extra_started_commands="reload save"
+command="/usr/bin/znc"
+command_args="--datadir ${ZNC_DATADIR}"
+pidfile="${ZNC_PIDFILE:-/run/znc/znc.pid}"
+user=${ZNC_USER:-znc}
+group=${ZNC_GROUP:-znc}
+start_stop_daemon_args="--chdir \"${ZNC_DATADIR}\" --user ${user} --group ${group} ${ZNC_SSDARGS}"
```

Comment:
Not sure what you mean: If znc is unable to start (configuration problem for example) it is normal that no pidfile will be created. But thanks to `ZNC_SSDARGS` which defaults to `--wait 1000` *start-stop-daemon* will *now* notice the startup failure.

If znc is able to start the pidfile should be written. If not verify that your tested *znc* was configured to use `--system-wide-config-as`.